mu/wheels/__init__.py

```diff
@@ -37,6 +41,7 @@ class WheelsBuildError(WheelsError):
     ("esptool", "esptool==3.*"),
 ]
 WHEELS_DIRPATH = os.path.dirname(__file__)
+ZIP_FILEPATH = os.path.join(WHEELS_DIRPATH, mu_version + ".zip")

```

Comment:
I'm having an issue with the `WHEELS_DIRPATH` not pointing to the final location where the wheels are downloaded, and so pip not finding a wheel previously downloaded to the temp path.
This is because I'm using Python 3.8 in macOS, and so it is triggering this workaround in the wheels module, which passes `WHEELS_DIRPATH` to pip:
https://github.com/mu-editor/mu/blob/4b83162f3cadf777f7ff978e4313ba19bde280a8/mu/wheels/__init__.py#L41-L57

You might have to edit the origina `mode_packages` to trigger this:
```python
mode_packages = [
    ("pygame", "pygame==2.0.1"),
    ("numpy", "numpy==1.20.1"),
    ("pgzero", "pgzero>=1.2.1", "--no-index", "--find-links=" + WHEELS_DIRPATH,),
    ("flask", "flask==1.1.2"),
    ("qtconsole", "qtconsole==4.7.4"),
    ("esptool", "esptool==3.*"),
]
```




[mu.log](https://github.com/mu-editor/mu/files/6682691/mu.log)
```
2021-06-20 17:11:02,462 - mu.wheels:89(pip_download) INFO: Running pip download for pygame / https://github.com/mu-editor/pygame/releases/download/2.0.1/pygame-2.0.1-cp38-cp38-macosx_10_9_intel.whl / []
2021-06-20 17:11:09,756 - mu.wheels:109(pip_download) DEBUG: Collecting pygame==2.0.1
  Downloading https://github.com/mu-editor/pygame/releases/download/2.0.1/pygame-2.0.1-cp38-cp38-macosx_10_9_intel.whl (6.9 MB)
  Saved /private/var/folders/jy/jkj1bwkd1blg0fd67553zct00000gn/T/tmpe5zh77f1/pygame-2.0.1-cp38-cp38-macosx_10_9_intel.whl
Successfully downloaded pygame
WARNING: You are using pip version 20.2.3; however, version 21.1.2 is available.
You should consider upgrading via the '/Users/microbit-carlos/workspace/mine/mu-upstream/.venv/bin/python -m pip install --upgrade pip' command.
2021-06-20 17:11:09,756 - mu.wheels:89(pip_download) INFO: Running pip download for numpy / numpy==1.20.1 / []
2021-06-20 17:11:14,693 - mu.wheels:109(pip_download) DEBUG: Collecting numpy==1.20.1
  Using cached numpy-1.20.1-cp38-cp38-macosx_10_9_x86_64.whl (16.0 MB)
  Saved /private/var/folders/jy/jkj1bwkd1blg0fd67553zct00000gn/T/tmpe5zh77f1/numpy-1.20.1-cp38-cp38-macosx_10_9_x86_64.whl
Successfully downloaded numpy
WARNING: You are using pip version 20.2.3; however, version 21.1.2 is available.
You should consider upgrading via the '/Users/microbit-carlos/workspace/mine/mu-upstream/.venv/bin/python -m pip install --upgrade pip' command.
2021-06-20 17:11:14,693 - mu.wheels:89(pip_download) INFO: Running pip download for pgzero / https://github.com/mu-editor/pgzero/releases/download/mu-wheel/pgzero-1.2-py3-none-any.whl / ['--no-index', '--find-links=/Users/microbit-carlos/workspace/mine/mu-upstream/mu/wheels']
2021-06-20 17:11:17,449 - mu.wheels:109(pip_download) DEBUG: Looking in links: /Users/microbit-carlos/workspace/mine/mu-upstream/mu/wheels
Collecting pgzero==1.2
  Downloading https://github.com/mu-editor/pgzero/releases/download/mu-wheel/pgzero-1.2-py3-none-any.whl (71 kB)
  Saved /private/var/folders/jy/jkj1bwkd1blg0fd67553zct00000gn/T/tmpe5zh77f1/pgzero-1.2-py3-none-any.whl
ERROR: Could not find a version that satisfies the requirement numpy (from pgzero==1.2) (from versions: none)
ERROR: No matching distribution found for numpy (from pgzero==1.2)
2021-06-20 17:11:17,452 - mu.virtual_environment:630(ensure_and_create) DEBUG: Emitter: <bound PYQT_SIGNAL display_text of StartupWorker object at 0x119ddeaf0>; Splash Handler; <SplashLogHandler (INFO)>
```

Reply:
Ok; thanks for the heads-up there. I'll have a look later to see the best way to address.

Reply:
I believe I've addressed this in https://github.com/tjguk/mu/commit/d6ffa3a8680ad33722aa7c5339be93ffdac5ce5e basically by always adding the temp download area to  `--find-links`. I've ad hoc tested it (by forcing the Windows build to fail in an equivalent way) but I don't have a test in place yet.

Reply:
Yeah, gave this a spin and works great, thanks Tim!